provider/resource_keycloak_realm.go

```diff
@@ -255,6 +255,63 @@ func resourceKeycloakRealm() *schema.Resource {
 					},
 				},
 			},
+
+			//Security Defenses
+			"security_defenses": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"headers": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"x_frame_options": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  "SAMEORIGIN",
+									},
+									"content_security_policy": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  "frame-src 'self'; frame-ancestors 'self'; object-src 'none';",
+									},
+									"content_security_policy_report_only": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  "",
+									},
+									"x_content_type_options": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  "nosniff",
+									},
+									"x_robots_tag": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  "none",
+									},
+									"x_xss_protection": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  "1; mode=block",
+									},
+									"strict_transport_security": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  "max-age=31536000; includeSubDomains",
```

Comment:
Should we be setting these defaults manually, or relying on Keycloak to do this for us?  I'm worried that there is a chance that upgrading your Keycloak instance could change the recommended defaults, but the provider would still set the old values.

Reply:
This allows users to only fill in the fields where they want to diverge from the default keycloak value. If there is no default value in the scheme, we would need to make every field required. This would mean that the user would need to go and look in keycloak to copy the values of the fields they do not want to change.
If we put the default on empty string, an empty string will passed to keycloak, and imho that is also not a desired result for the users.

Even if keycloak would change the recommanded defaults in the future, these current default values will be better than no value, or a value that the user would have copied from keycloak form.

Reply:
I agree that we would want sensible defaults to be set if the user omits any of these attributes, but what I'm suggesting is to let Keycloak handle that instead of hardcoding the defaults (which may change in the future) within the provider.

If a user doesn't specify one of these attributes, we don't have to send Keycloak an empty string - we can use `omitempty` within the JSON definition for the struct to tell the provider to not include properties with empty string values when sending the request to Keycloak.  This will allow Keycloak to set the default to whatever it wants.

What are your thoughts on this?

Reply:
I can remove the defaults.
But Imho I prefer to have the defaults, because of following example without the defaults:



lets focus on the x_robots_tag field.

User creates following config:

```
resource "keycloak_realm" "test" {
	realm        = "test"
	enabled      = true
	display_name = "foo"
	
	security_defenses {
		headers {
			x_frame_options = "DENY"
		}
	}
}
```
in keycloak x_robots_tag has now the value none

user changes the config to:

```
resource "keycloak_realm" "test" {
	realm        = "test"
	enabled      = true
	display_name = "foo"

	security_defenses {
		headers {
			x_frame_options = "DENY"
			x_robots_tag = "bla"
		}
	}
}
```

in keycloak x_robots_tag has now the value bla

user removes the value again

```
resource "keycloak_realm" "test" {
	realm        = "test"
	enabled      = true
	display_name = "foo"
	
	security_defenses {
		headers {
			x_frame_options = "DENY"
		}
	}
}
```
in keycloak x_robots_tag has now the value ""

because

```
Terraform will perform the following actions:

  ~ keycloak_realm.test
      security_defenses.0.headers.0.x_robots_tag: "bla" => ""
```


In the end this heavily depends on what you decide on the comment/discussion below. Because If we expect that the plugin restores the default values, we need the default values in the scheme. If not, we can remove the defaults.

Reply:
Okay, I'm fine with keeping what you currently have implemented.